gps-rtk-serial-no-network/gps_rtk_serial_no_network.go

```diff
@@ -408,30 +413,31 @@ func (g *rtkSerialNoNetwork) Readings(ctx context.Context, extra map[string]inte
 // Close shuts down the RTKSerialNoNetwork.
 func (g *rtkSerialNoNetwork) Close(ctx context.Context) error {
 	g.cancelFunc()
+	g.activeBackgroundWorkers.Wait()
 
 	g.correctionReaderMu.Lock()
 
-	//close the reader
+	// close the reader.
 	if g.correctionReader != nil {
 		if err := g.correctionReader.Close(); err != nil {
 			g.correctionReaderMu.Unlock()
-			return err
+			g.err.Set(err)
+			g.logger.Errorf("failed to close correction reader %s", err)
```

Comment:
It will return the set error below this on line 441, but thought that if it fails to close the correctionReader it should still try to close the writer rather than just return.